admin/UpdateDatabasePrivileges.pl

```diff
@@ -72,40 +92,39 @@
           FROM pg_namespace
          WHERE nspname = any(?)
         SQL
-        [@FULL_SCHEMA_LIST],
+        [@schemas],
     );
     for my $schema (@$existing_schemas) {
         my $quoted_schema = $sql->dbh->quote_identifier($schema);
-        log_info {
-            "Updating privileges for $quoted_primary_role on $quoted_schema tables in $database"
-        };
         my $revoked_privileges;
         if ($grant_privileges) {
-            $sql->do(<<~"SQL");
-                GRANT USAGE
-                   ON SCHEMA $quoted_schema
-                   TO $quoted_primary_role;
-                SQL
+            $log_and_do_query->(
+                "GRANT USAGE ON SCHEMA $quoted_schema " .
```

Comment:
BTW, I intentionally used string concatenation rather than `<<~"SQL"` for these so that the queries are logged on a single line.